### ISSUE TITLE
How to use embedding to compute triplet score

### ISSUE THREAD

**Issue by zhongpeixiang** [DeepGraphLearning/KnowledgeGraphEmbedding#17]
I don't quite understand your training script about head-batch. Anyway, I have pretrained ent and rel embeddings without understanding it. 

Now I want to figure out that given a triplet: ```h [x1, ..., x200], r=[r1, ..., r100] and t=[y1, ..., y200]```, how to compute its score?

A detailed description will be much appreciated.

**Comment by Edward-Sun** [comment]
Hi, Peixiang,

For RotatE, you can first transform relation embedding into phases like:

https://github.com/DeepGraphLearning/KnowledgeGraphEmbedding/blob/2442e8b9bb1a40e44edabcb3cc4f145ac24b1c7b/codes/model.py#L208

where self.embedding_range.item() is model-dependent. Then you can calculate the score by either head-batch mode or tail-batch mode, they are equivalent for a single triplet.

For example, in batch-mode, you can write something like:

re_relation = torch.cos(phase_relation)
im_relation = torch.sin(phase_relation)
re_score = re_relation * re_tail + im_relation * im_tail - re_head
im_score = re_relation * im_tail - im_relation * re_tail - im_head
score = torch.stack([re_score, im_score], dim = 0)
score = score.norm(dim = 0)
score = - score.sum(dim = 2)
